tests/projects/backend/test_local.py

```diff
@@ -4,67 +4,75 @@
+def test_docker_s3_artifact_cmd_and_envs_from_home(monkeypatch):
     mock_env = {}
-    with mock.patch.dict("os.environ", mock_env), mock.patch(
-        "posixpath.exists", return_value=True
-    ), mock.patch("posixpath.expanduser", return_value="mock_volume"):
+    monkeypatch.setenvs(mock_env)
+    with mock.patch("posixpath.exists", return_value=True), mock.patch(
+        "posixpath.expanduser", return_value="mock_volume"
+    ):
         cmds, envs = _get_docker_artifact_storage_cmd_and_envs("s3://mock_bucket")
         assert cmds == ["-v", "mock_volume:/.aws"]
         assert envs == mock_env
 
 
-def test_docker_wasbs_artifact_cmd_and_envs_from_home():
+def test_docker_wasbs_artifact_cmd_and_envs_from_home(monkeypatch):
     mock_env = {
         "AZURE_STORAGE_CONNECTION_STRING": "mock_connection_string",
         "AZURE_STORAGE_ACCESS_KEY": "mock_access_key",
     }
     wasbs_uri = "wasbs://container@account.blob.core.windows.net/some/path"
-    with mock.patch.dict("os.environ", mock_env), mock.patch(
-        "azure.storage.blob.BlobServiceClient"
-    ):
+    monkeypatch.setenvs(mock_env)
+    with mock.patch("azure.storage.blob.BlobServiceClient"):
         cmds, envs = _get_docker_artifact_storage_cmd_and_envs(wasbs_uri)
         assert cmds == []
         assert envs == mock_env
 
 
-def test_docker_gcs_artifact_cmd_and_envs_from_home():
+def test_docker_gcs_artifact_cmd_and_envs_from_home(monkeypatch):
     mock_env = {
         "GOOGLE_APPLICATION_CREDENTIALS": "mock_credentials_path",
     }
     gs_uri = "gs://mock_bucket"
-    with mock.patch.dict("os.environ", mock_env, clear=True):
-        cmds, envs = _get_docker_artifact_storage_cmd_and_envs(gs_uri)
-        assert cmds == ["-v", "mock_credentials_path:/.gcs"]
-        assert envs == {"GOOGLE_APPLICATION_CREDENTIALS": "/.gcs"}
+    monkeypatch.delenvs(["GOOGLE_APPLICATION_CREDENTIALS"], raising=False)
+    monkeypatch.setenvs(mock_env)
+    cmds, envs = _get_docker_artifact_storage_cmd_and_envs(gs_uri)
+    assert cmds == ["-v", "mock_credentials_path:/.gcs"]
+    assert envs == {"GOOGLE_APPLICATION_CREDENTIALS": "/.gcs"}
 
 
-def test_docker_hdfs_artifact_cmd_and_envs_from_home():
+def test_docker_hdfs_artifact_cmd_and_envs_from_home(monkeypatch):
     mock_env = {
         "MLFLOW_KERBEROS_TICKET_CACHE": "/mock_ticket_cache",
         "MLFLOW_KERBEROS_USER": "mock_krb_user",
         "MLFLOW_PYARROW_EXTRA_CONF": "mock_pyarrow_extra_conf",
     }
     hdfs_uri = "hdfs://host:8020/path"
-    with mock.patch.dict("os.environ", mock_env, clear=True):
-        cmds, envs = _get_docker_artifact_storage_cmd_and_envs(hdfs_uri)
-        assert cmds == ["-v", "/mock_ticket_cache:/mock_ticket_cache"]
-        assert envs == mock_env
+    monkeypatch.delenvs(
+        [
+            "MLFLOW_KERBEROS_TICKET_CACHE",
+            "MLFLOW_KERBEROS_USER",
+            "MLFLOW_PYARROW_EXTRA_CONF",
+        ],
+        raising=False,
+    )
```

Comment:
this is unnecessary

Reply:
I thought `clear=True` in mock.patch.dict intended to delete envs specified in arguments

Changes were based on my [first commit](https://github.com/mlflow/mlflow/commit/70fab1b608216c9579687534232cabcbbcfe3927#diff-55cacde23edacba590f68eaa71f49ab1591adf3139965bc500d1a62971d3224cL290)  in `test_conda_path()` test:
`with mock.patch.dict("os.environ", mock_env, clear=True):` -> `monkeypatch.delenvs(["CONDA_EXE", MLFLOW_CONDA_HOME], raising=False)`